webapp/src/components/Quiz/Quiz.js

```diff
@@ -1,144 +1,254 @@
-import React, { useState, useRef } from 'react'
-import Countdown, { zeroPad } from 'react-countdown'
+import "./Quiz.css";
+import React, { useState, useRef, useEffect } from "react";
+import Countdown, { zeroPad } from "react-countdown";
+import Question from "./Question";
+import FinalResult from "../FinalResult/FinalResult";
+import CircularProgress from '@mui/material/CircularProgress';
+import { ReactComponent as StopwatchIcon } from "../../assets/stopwatch-solid.svg";
+import { useAuth } from '../../hooks/useAuth'
+
+const Quiz = ({ timerValue, level }) => {
+  const apiEndpoint = "http://localhost:8000";
+
+  const [questions, setQuestions] = useState([]);
+  const [currentQuestionIndex, setCurrentQuestionIndex] = useState(0);
+  const [quizData, setQuizData] = useState({});
+  const [haveQuestions, setHaveQuestions] = useState(false);
+  const [isFinished, setIsFinished] = useState(false);
+  const [score, setScore] = useState(0);
+  const [btnDisabled, setBtnDisabled] = useState(false);
+  const [timer, setTimer] = useState(Date.now() + timerValue * 1000);
+  const [timerIndex, setTimerIndex] = useState(0);
+  const { user } = useAuth()
+  const countdownTimer = useRef();
+
+  useEffect(() => {
+    (async () => {
+      if (!haveQuestions) {
+        setQuestions(await getQuestions());
+        setHaveQuestions(true);
+      }
+    })();
+  }, [haveQuestions]);
+
+  useEffect(() => {
+    if (questions) {
+      setTimer(Date.now() + timerValue * 1000);
+    }
+  }, [questions]);
+
+  useEffect(() => {
+    if (isFinished) {
+      //updateStatistics();
```

Comment:
Yes, I left it commented on purpose because it was giving me problems and I had to look at it. Now the function works perfectly and updates the statistics in each user's game.